+++
author = "SOMRC Staff"
description = ""
title = "Bioinformatics & Genomics"
date = "2017-01-31T09:48:06-05:00"
draft = false
tags = ["bioinformatics","genomics"]
categories = ["services"]
images = [""]

+++

<p class=lead>School of Medicine Research Computing (SOMRC) provides state-of-the-art resources and expertise in handling and analyzing genomics and bioinformatics data. Bioinformatics is a quickly evolving field with new biological and computational techniques being formalized and adopted at a fast pace. Hence, the following is only a brief cross-section of the ways researchers can use SOM RCâ€™s expertise and computing resources for their bioinformatics research. Please contact the SOM RC with your computational research needs.</p>

<hr size=1 />

# Bioinformatics tools
-	Running genomic alignment and analysis tools on desktop or on Rivanna, Ivy, or AWS including Bowtie, Samtools, Bedtools, MACS, HTSeq, and TopHat
-	Accessing and retrieving data from ArrayExpress, GEO, and other databases

# ChIP-Seq, ChIP-chip, RNA-Seq and other NGS analyses
-	Formalizing an automated analysis pipeline in Linux 
-	Background subtraction, normalization, and processing of the data
-	Statistical modeling, regression, and computational methods for exploratory analyses (clustering and heat maps), significance testing (e.g., statistical tests), and hypothesis testing (e.g., proportional hazards models) 
-	Quantitative and correlative analyses with other publicly available datasets for biological insight
 
# AWS Cloud
-	Setting up ChIP-Seq/RNA-Seq analysis pipelines on AWS (or Ivy and Rivanna). This is an extensive area, for some examples check out the items below: 
-	Setting up the [StarCluster](http://star.mit.edu/cluster/docs/latest/overview.html) on [AWS](http://www.oliverelliott.org/article/computing/tut_bio_aws/) to run bioinformatics analyses:
-	[Omics Pipe](http://omics-pipe.readthedocs.io/en/latest/about.html) (for RNA-Seq, ChIP-Seq) that utilizes a StarCluster
-	Running [Bcbio-nextgen](http://bcbio-nextgen.readthedocs.io/en/latest/contents/pipelines.html) pipeline on AWS for RNA-Seq, cancer variant calling, and ChIp-Seq.

# Ivy and Rivanna High Performance Computing
Ivy is a high performance computing resource that can be used for HIPAA compliant translational research. Amongst other things, the SOMRC team can help with:
-	Setting up alignment, variant calling, and analysis pipelines for ChIP-Seq, ChIP-chip, RNA-Seq and other genomic experiments on Rivanna and Ivy. 
-	Using [GATK](https://software.broadinstitute.org/gatk/guide/article.php?id=1988) on Ivy or Rivanna employing multi-threading and scatter-gather for fast SNP variant calling.

# Long-term collaborations
We warmly welcome long-term collaborations with experimentalists and computational biologists. Working with experimental biologists we can chart out experimental design, come up with controls, and think about statistical and computational analyses so the experiment is designed to extract the most value from the data. With computational biologists, we can build and maintain code and pipelines, as well as test new algorithms on multi-core and multi-node machines.


# New Research Lines
We are actively exploring and researching new computational techniques that can reap big rewards in genomics and bioinformatics. Our particular focus is on machine learning, especially deep learning, and big data architectures, especially MapReduce and Spark.
