+++
author = "SOMRC Staff"
description = ""
date = "2017-03-23T17:17:35-04:00"
title = "Basic Science Projects"
draft = false
tags = ["collaborations", "basic science"]
categories = ["projects"]
images = [""]

+++

<div class="bd-callout bd-callout-warning">
<p class=lead>School of Medicine Research Computing is engaged in multiple collaborative projects in support of basic science research. Below is a list of some recent collaborations in this area.</p>
</div>

# PHACTR1 and Smooth Muscle Cell Behavior

Genome wide associate studies (GWAS) have revealed a number of loci that are associated with increased risk for Coronary Artery Disease (CAD). SOMRC is helping analyze phenotypic data collected in vitro in relation to PHACTR1, which is one of the genes associated with CAD. So far this analysis has included interactive visualizations, eQTL plots, merging of genotype and phenotype data and statistical hypothesis testing.

For a more general description of this collaboration, refer [here](https://somrc.virginia.edu/project/cardiovascular-genomics/). 

**Redouane Aherrarou ([Center for Public Health Genomics](https://med.virginia.edu/cphg/))**

- - -

# Relative Proportion of Sites with Intermediate Methylation (RPIM)

SOMRC is working with researchers in the Center for Public Health Genomics to write an R package to calculate RPIM scores, which represent the epigenetic heterogeneity in a bisulfite sequencing sample.

**Nathan Sheffield ([Center for Public Health Genomics](https://med.virginia.edu/cphg/))**

- - -

# Transcription factor-chromatin Binding Dynamics

Two important measures of the in vivo interaction of transcription factors with chromatin are the search time and the residence time. The former refers to the time it takes a factor to find its binding location, while the latter is the time the factor physically attaches to the chromatin. By quantifying the interaction dynamics of transcription factors, researchers hope to understand the role of these factors in basic cellular processes such as transcription and gene regulation. The SOMRC team is working with collaborators from UVA and the NIH to understand the dynamics of the Gal4 protein in yeast. The project involves quantitatively analyzing ChIP-qPCR data, writing and running non-linear regression and statistical routines in Mathematica, and developing numerical simulations to determine the error bounds on the kinetic parameters. 

**Stefan Bekiranov ([Biochemistry and Molecular Genetics](https://bmg.med.virginia.edu))**

- - -

# Functional Connectome Fingerprinting

Functional magnetic resonance imaging (fMRI) can be used to assess functional activity in the brain and connectivity between different regions of interest (ROIs), and a functional connectome is a map of the interactions between ROIs. Previous research has shown that a functional connectome contains enough unique characteristics, not unlike a fingerprint, that it can be used for accurate identification of an individual subject from a large group. SOMRC is working with the UVA Functional Neuroradiology Lab to perform this fingerprinting analysis for a wide variety of populations and to develop innovative ways to visualize the results.

**Jason Druzgal ([Radiology and Medical Imaging](http://druzgallab.com))**

- - - 

# Microbiome Analysis of Hospital Sink Drains

Sink drains are notoriously characterized as reservoirs of pathogens causing nosocomial transmissions in hospitals worldwide. Outbreaks where sinks have been implicated as source of antibiotic resistant bacteria have upsurged over the last few years. To understand transmission dynamics University of Virginia School of Medicine has established a unique "Sink Lab" for this research. This one-of-the kind laboratory establishes UVa as worldwide frontrunners in investigating sink related antibiotic resistant bacteria and how they spread. SOMRC is working with the UVa Sink Lab for genomic analysis of the sink biomass. 

**PI: Amy Mathers ([Infectious Diseases](https://med.virginia.edu/infectious-diseases/) & [UVa Sink Lab](http://uvasinklab.org/))**

- Comparative genomic analysis of gram-negative bacterial isolates –  
    The analysis aims at tracking the mobile genetic element blaKPC gene, which encodes for Klebsiella pneumoniae carbapenemase (KPC) enzyme that confers resistance to all beta lactam agents including penicillins, cephalosporins, monobactams and carbapenems. As a part of this project, whole-genome shotgun sequencing data for about 1500 samples will be analyzed to assess the risk of acquisition of Carbapenemase producing Enterobacteriaceae from exposure to contaminated waste water premise plumbing.   
- Metagenomic analysis –  
    This project, under a contract for the Center for Disease Control and Prevention (CDC),  aims at understanding the temporal dynamics of hospital sink microbiome. Taxonomic and functional analysis of whole metagenomic shotgun sequencing data from longitudinal sampling will shed light on the transfer and sustenance of high-risk antibiotic resistance genes in the hospital environments.
