+++
author = "SOMRC Staff"
description = ""
date = "2017-03-23T17:17:35-04:00"
title = "Basic Science Projects"
draft = false
tags = ["collaborations", "basic science"]
categories = ["projects"]
images = [""]

+++

<p class=lead>School of Medicine Research Computing is engaged in multiple collaborative projects in support of basic science research. Below is a list of some recent collaborations in this area.</p>

- - -

# PHACTR1 and Smooth Muscle Cell Behavior

Genome wide associate studies (GWAS) have revealed a number of loci that are associated with increased risk for Coronary Artery Disease (CAD). SOMRC is helping analyze phenotypic data collected in vitro in relation to PHACTR1, which is one of the genes associated with CAD. So far this analysis has included interactive visualizations, eQTL plots, merging of genotype and phenotype data and statistical hypothesis testing.

For a more general description of this collaboration, refer [here](https://somrc.virginia.edu/project/cardiovascular-genomics/). 

**Redouane Aherrarou ([Center for Public Health Genomics](https://med.virginia.edu/cphg/))**

- - -

# Relative Proportion of Sites with Intermediate Methylation (RPIM)

SOMRC is working with researchers in the Center for Public Health Genomics to write an R package to calculate RPIM scores, which represent the epigenetic heterogeneity in a bisulfite sequencing sample.

**Nathan Sheffield ([Center for Public Health Genomics](https://med.virginia.edu/cphg/))**

- - -

# Transcription factor-chromatin Binding Dynamics

Two important measures of the in vivo interaction of transcription factors with chromatin are the search time and the residence time. The former refers to the time it takes a factor to find its binding location, while the latter is the time the factor physically attaches to the chromatin. By quantifying the interaction dynamics of transcription factors, researchers hope to understand the role of these factors in basic cellular processes such as transcription and gene regulation. SOMRC is working with collaborators from UVA and the NIH on extending a statistical physics model to quantify the search and residence times for Gal4 protein with yeast chromatin. The SOMRC team is helping researchers quantitatively analyze ChIP-qPCR data, writing and running non-linear regression and statistical routines in Mathematica, and developing numerical simulations to determine the error bounds on the kinetic parameters.

**Stefan Bekiranov ([Biochemistry and Molecular Genetics](https://bmg.med.virginia.edu))
