+++
author = "SOMRC Staff"
description = ""
title = "Bioinformatics Software on Rivanna"
date = "2017-01-31T09:48:06-05:00"
draft = false
tags = ["bioinformatics","genomics", "rivanna","tools"]
categories = ["services"]
images = [""]
+++

<p class=lead>UVA research community has access to 100+ bioinformatics software installed and ready-to-use on Rivanna. They are all available via the LMod module system.</p>

[Click here](https://arcs.virginia.edu/software-list) for a comprehensive list.  

<hr size=1 />
 
**BEDTools** - BEDTools utilities allow one to intersect, merge, count, complement, and shuffle genomic intervals from multiple files in widely-used genomic file formats such as BAM, BED, GFF/GTF, VCF.

**BLAST+** - BLAST+ is a suite of command-line tools that offers applications for BLAST search, BLAST database creation/examination, and sequence filtering. 

**[BWA](https://discuss.rc.virginia.edu/t/ngs-alignment-using-bwa/859)** - BWA is a software package for mapping low-divergent sequences against a large reference genome, such as the human genome. It consists of three algorithms: BWA-backtrack, BWA-SW and BWA-MEM 

**[Bowtie2](https://discuss.rc.virginia.edu/t/how-to-run-bowtie2-on-rivanna/126)** - Bowtie2 is a memory-efficient tool for aligning short sequences to long reference genomes.

**FastQC** - FastQC is a Java application that generates a comprehensive quality control report for raw sequencing data.  

**GATK** - The Genome Analysis Toolkit provide tools for variant discovery. In addition to SNP and INDEL identification in germline DNA and RNAseq data, GATK tools include somatic short variant calling, as well as tackle copy number and structural variation.  

**Picard** - Picard is a set of command line tools for manipulating high-throughput sequencing (HTS) data and formats such as SAM/BAM/CRAM and VCF.

**SAMTools** - SAMTools provide various utilities for manipulating alignments in the SAM format, including sorting, merging, indexing and generating alignments in a per-position format.

**SPAdes** - SPAdes provide pipelines for assembling genomes from Illumina and IonTorrent reads, as well as hybrid assemblies using PacBio, Oxford Nanopore and Sanger reads. It supports paired-end reads, mate-pairs and unpaired reads.  

**STAR2** - Spliced Transcripts Alignment to a Reference (STAR) is a RNA-seq aligner based on an algorithm that uses sequential maximum mappable seed search in uncompressed suffix arrays followed by seed clustering and stitching procedure.

**vsearch** - VSEARCH (stands for Vectorized Search) is a toolkit for nucleotide sequence analyses, including database search and clustering algorithms. It supports clustering, chimera detection, database searching, merging of paired-end reads, and other sequence manipulation tools.

