+++
author = "SOMRC Staff"
description = ""
title = "Bioinformatics Software on Rivanna"
date = "2017-01-31T09:48:06-05:00"
draft = false
tags = ["bioinformatics","genomics", "rivanna","tools"]
categories = ["services"]
images = [""]
+++

<p class=lead>UVA research community has access to 100+ bioinformatics software installed and ready-to-use on Rivanna. They are all available via the LMod module system.
<br/>
<br/>
<a href="https://arcs.virginia.edu/software-list" target="blank">Click here</a> for a comprehensive list.
</p>

<hr size=1 />

<h2>Popular Bioinformatics Software</h2>

Below are some popular tools and useful links for their documentation and usage:

<style type="text/css">
.tg  {border-collapse:collapse;border-spacing:0;border-color:#ccc;}
.tg td{font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:0px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333
;background-color:#fff;}
.tg th{font-family:Arial, sans-serif;font-size:14px;font-weight:normal;padding:10px 5px;border-style:solid;border-width:0px;overflow:hidden;word-break:normal;border-co
lor:#ccc;color:#333;background-color:#f0f0f0;}
.tg .tg-hy9w{background-color:#eceeef;border-color:inherit;vertical-align:middle}
.tg .tg-dc35{background-color:#f9f9f9;border-color:inherit;vertical-align:middle}
.tg .tg-hy9w-nw{background-color:#eceeef;border-color:inherit;vertical-align:middle;white-space:nowrap}
.tg .tg-dc35-nw{background-color:#f9f9f9;border-color:inherit;vertical-align:middle;white-space:nowrap}
.tg .tg-0qmj{font-weight:bold;background-color:#eceeef;border-color:inherit;vertical-align:middle}
</style>
<div>
<table class="tg">
  <tr>
    <th class="tg-0qmj">Tool</th>
    <th class="tg-0qmj">Version</th>
    <th class="tg-0qmj">Description</th>
    <th class="tg-0qmj">Useful Links</th>
  </tr>
  <tr>
    <td class="tg-dc35"><font color="#bd4147">BEDTools</font></td>
    <td class="tg-dc35">2.26.0</td>
    <td class="tg-dc35">BEDTools utilities allow one to intersect, merge, count, complement, and shuffle genomic intervals from multiple files in widely-used genomic file formats such as BAM, BED, GFF/GTF, VCF.</td>
	<td class="tg-dc35-nw">
      <a href="http://bedtools.readthedocs.io/en/latest/" target="blank" style="color:#0275d8">Homepage</a><br/>
      <a href="http://quinlanlab.org/tutorials/bedtools/bedtools.html" target="blank" style="color:#0275d8">Tutorial</a><br/>
    </td>
  </tr>
  <tr>
    <td class="tg-hy9w"><font color="#bd4147">BLAST+</font></td>
    <td class="tg-hy9w">2.7.1</td>
    <td class="tg-hy9w">BLAST+ is a suite of command-line tools that offers applications for BLAST search, BLAST database creation/examination, and sequence filtering.</td>
	<td class="tg-hy9w-nw">
      <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" target="blank" style="color:#0275d8">Web BLAST</a><br/>
      <a href="https://www.ncbi.nlm.nih.gov/books/NBK279690/" target="blank" style="color:#0275d8">Manual</a><br/>
    </td>
  </tr>
  <tr>
    <td class="tg-dc35"><font color="#bd4147">BWA</font></td>
    <td class="tg-dc35">0.7.17</td>
    <td class="tg-dc35">BWA is a software package for mapping low-divergent sequences against a large reference genome, such as the human genome. It consists of three algorithms: BWA-backtrack, BWA-SW and BWA-MEM</td>
	<td class="tg-dc35-nw">
      <a href="http://bio-bwa.sourceforge.net/" target="blank" style="color:#0275d8">Homepage</a><br/>
      <a href="http://bio-bwa.sourceforge.net/bwa.shtml" target="blank" style="color:#0275d8">Manual</a><br/>
      <a href="https://discuss.rc.virginia.edu/tags/bwa" target="blank" style="color:#0275d8">Rivanna FAQs</a><br/>
    </td>
  </tr>
  <tr>
    <td class="tg-hy9w"><font color="#bd4147">Bowtie2</font></td>
    <td class="tg-hy9w">2.2.9</td>
    <td class="tg-hy9w">Bowtie2 is a memory-efficient tool for aligning short sequences to long reference genomes.</td>
	<td class="tg-hy9w-nw">
      <a href="http://bowtie-bio.sourceforge.net/index.shtml" target="blank" style="color:#0275d8">Homepage</a><br/>
      <a href="http://http://bowtie-bio.sourceforge.net/manual.shtml" target="blank" style="color:#0275d8">Manual</a><br/>
      <a href="https://discuss.rc.virginia.edu/tags/bowtie2" target="blank" style="color:#0275d8">Rivanna FAQs</a><br/>
  </tr>
  <tr>
    <td class="tg-dc35"><font color="#bd4147">FastQC</font></td>
    <td class="tg-dc35">0.11.5</td>
    <td class="tg-dc35">FastQC is a Java application that generates a comprehensive quality control report for raw sequencing data.</td>
	<td class="tg-dc35-nw">
      <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="blank" style="color:#0275d8">Homepage</a><br/>
      <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/" target="blank" style="color:#0275d8">Documentation</a><br/>
    </td>
  </tr>
  <tr>
    <td class="tg-hy9w"><font color="#bd4147">GATK</font></td>
    <td class="tg-hy9w">4.0.0.0</td>
    <td class="tg-hy9w">The Genome Analysis Toolkit provide tools for variant discovery. In addition to SNP and INDEL identification in germline DNA and RNAseq data, GATK tools include somatic short variant calling, as well as tackle copy number and structural variation.</td>
	<td class="tg-hy9w-nw">
      <a href="https://software.broadinstitute.org/gatk/documentation/" target="blank" style="color:#0275d8">User Guide</a><br/>
      <a href="https://discuss.rc.virginia.edu/tags/gatk" target="blank" style="color:#0275d8">Rivanna FAQs</a><br/>
  </tr>
  <tr>
    <td class="tg-dc35"><font color="#bd4147">Picard</font></td>
    <td class="tg-dc35">2.1.1</td>
    <td class="tg-dc35">Picard is a set of command line tools for manipulating high-throughput sequencing (HTS) data and formats such as SAM/BAM/CRAM and VCF.</td>
	<td class="tg-dc35-nw">
      <a href="https://broadinstitute.github.io/picard/" target="blank" style="color:#0275d8">Homepage</a><br/>
      <a href="https://broadinstitute.github.io/picard/command-line-overview.html" target="blank" style="color:#0275d8">Documentation</a><br/>
    </td>
  </tr>
  <tr>
    <td class="tg-hy9w"><font color="#bd4147">SAMTools</font></td>
    <td class="tg-hy9w">1.7</td>
    <td class="tg-hy9w">SAMTools provide various utilities for manipulating alignments in the SAM format, including sorting, merging, indexing and generating alignments in a per-position format.</td>
	<td class="tg-hy9w-nw">
      <a href="http://samtools.sourceforge.net/" target="blank" style="color:#0275d8">Homepage</a><br/>
      <a href="http://www.htslib.org/doc/samtools.html" target="blank" style="color:#0275d8">Manual</a><br/>
  </tr>
  <tr>
    <td class="tg-dc35"><font color="#bd4147">SPAdes</font></td>
    <td class="tg-dc35">3.10.1</td>
    <td class="tg-dc35">SPAdes provide pipelines for assembling genomes from Illumina and IonTorrent reads, as well as hybrid assemblies using PacBio, Oxford Nanopore and Sanger reads. It supports paired-end reads, mate-pairs and unpaired reads. </td>
	<td class="tg-dc35-nw">
      <a href="http://bioinf.spbau.ru/spades" target="blank" style="color:#0275d8">Homepage</a><br/>
      <a href="http://spades.bioinf.spbau.ru/release3.11.1/manual.html" target="blank" style="color:#0275d8">Manual</a><br/>
    </td>
  </tr>
  <tr>
    <td class="tg-hy9w"><font color="#bd4147">STAR</font></td>
    <td class="tg-hy9w">2.5.3a</td>
    <td class="tg-hy9w">Spliced Transcripts Alignment to a Reference (STAR) is a RNA-seq aligner based on an algorithm that uses sequential maximum mappable seed search in uncompressed suffix arrays followed by seed clustering and stitching procedure.</td>
	<td class="tg-hy9w-nw">
      <a href="https://github.com/alexdobin/STAR" target="blank" style="color:#0275d8">Homepage</a><br/>
  </tr>
  <tr>
    <td class="tg-dc35"><font color="#bd4147">vsearch</font></td>
    <td class="tg-dc35">2.7.1</td>
    <td class="tg-dc35">VSEARCH (stands for Vectorized Search) is a toolkit for nucleotide sequence analyses, including database search and clustering algorithms. It supports clustering, chimera detection, database searching, merging of paired-end reads, and other sequence manipulation tools.</td>
	<td class="tg-dc35-nw">
      <a href="https://github.com/torognes/vsearch" target="blank" style="color:#0275d8">Homepage</a><br/>
    </td>
  </tr>
</table>

<hr size=1 />

<h2>Additional Support</h2>

We are continually extending our library of code templates for usage on Rivanna. If you wish to use a bioinformatics software for your project (not in the above list) and would like to see a _HowTo Use on Rivanna_ guide, create a topic on the UVA Research Computing [discussion board](https://discuss.rc.virginia.edu/). 


